src/IDCodeValidator.java

```diff
@@ -0,0 +1,166 @@
+                && isDayNumberCorrect(personalCode.dayNumber)
+                && isMonthNumberCorrect(personalCode.monthNumber)
+                && isBirthDateCorrect(getFullYear(personalCode.genderNumber, personalCode.yearNumber), personalCode.monthNumber, personalCode.dayNumber)
+                && checkControlNumber(idCode)){
+
+            System.out.println("ID is correct");
+            return true;
+        }
+        if (idCode.length() != 11) {
+            System.out.println("The length is not correct");
+
+        }
+
+        return false;
+    }
+
+
+    public static boolean isGenderNumberCorrect(int genderNumber) {
+        if (genderNumber == 1 || genderNumber == 3 || genderNumber == 5 || genderNumber == 2 || genderNumber == 4 || genderNumber == 6) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public static boolean isDayNumberCorrect(int dayNumber) {
+        if (dayNumber > 0 && dayNumber <= 31) {
+            System.out.println(dayNumber);
+            return true;
+        }
+        System.out.println("There is no such day");
+        return false;
+    }
+
+    public static boolean isMonthNumberCorrect(int monthNumber) {
+        if (monthNumber > 0 && monthNumber <= 12) {
+            System.out.println(monthNumber);
+            return true;
+        }
+        System.out.println("There is no such month");
+        return false;
+    }
+
+
+    public static boolean isBirthDateCorrect(int yearNumber, int monthNumber, int dayNumber) {
+        boolean result;
+        int[] numbersOfDays = {0, 31, 28, 31, 30, 31, 30, 31, 30, 31, 30, 31};
+        if (numbersOfDays[2] == 28 && isLeapYear(yearNumber)) {
+            result = (dayNumber <= numbersOfDays[monthNumber] + 1);
+        } else {
+            result = (dayNumber <= numbersOfDays[monthNumber]);
+        }
+        if (result) {
+            System.out.println("Your birth date is correct");
+        } else {
+            System.out.println("Your birth date is not correct");
+        }
+        return result;
+    }
+
+
+    public static boolean isLeapYear(int yearNumber) {
+        if ((yearNumber % 4 == 0 && yearNumber % 100 != 0) || (yearNumber % 400 == 0)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    public static boolean checkControlNumber(String idCode) {
+        int result = 0;
+        List<Integer> a = new ArrayList<>();
+        a.add(Integer.valueOf(idCode));
+        for (int i = 0; i < idCode.length(); i++) {
+            String n = String.valueOf(idCode.charAt(i));
+            a.add(Integer.parseInt(n));
+
+
+            List<Integer> x = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 1);
+            result = result + (a.get(i) * x.get(i));
+
+            int leftover = result % 11;
+
+
+            List<Integer> k = List.of(3, 4, 5, 6, 7, 8, 9, 1, 2, 3);
+
+            result = result + (a.get(i) * k.get(i));
+
+        }
```

Comment:
You need two separate loops, otherwise the overall result is going to be incorrect.
In the first loop you calculate the result with the weights from the first list.
Then you check for the remainder and see two things

-  IF remainder < 10 ->  then we check if the remainder is equal to the last digit of id code. If they are equal, then control number is correct, otherwise not
-  IF remainder  == 10 ->  then we do another loop where we calculate the sum again with the second set of weights (List.of(3, 4, 5, 6, 7, 8, 9, 1, 2, 3)). Then we also calculate remainder and see again two things. 

-  - if remainder < 10 -> then we check if the remainder is equal to the last digit of id code. If they are equal, then control number is correct, otherwise not
-  - if remainder  == 10 then we check if the last digit is equal to 0. If it is then control number is correct, otherwise not